components/esp_insights/src/esp_insights.c

```diff
@@ -153,6 +154,25 @@ static void esp_insights_common_cb(TimerHandle_t handle)
     }
 }
 
+static esp_err_t esp_insights_unregister_periodic_handler(void)
+{
+    esp_err_t err;
+
+    if (periodic_insights_entry->timer) {
+        ESP_LOGI(TAG, "Deleting the periodic timer");
+        if (xTimerDelete(periodic_insights_entry->timer, 10) != pdPASS) {
+            ESP_LOGE(TAG, "Failed to delete the periodic timer");
+        }
+    }
+
+    if ((err = esp_rmaker_work_queue_stop()) != ESP_OK)
+        ESP_LOGE(TAG, "Failed to stop esp_rmaker_work_queue: %s", esp_err_to_name(err));
+
+    free(periodic_insights_entry);
+    periodic_insights_entry = NULL;
+    return esp_rmaker_work_queue_deinit();
```

Comment:
This should be part of `esp_insights_deinit()` as `esp_rmaker_work_queue_init()` is called from `esp_insights_init()`.